### PR TITLE
Friends

### PR DESCRIPTION
**Pull request by Edith2019**
HI everyone,

Please have a look at the components: OtherProfile.js/ FindPeople.js and FriendsButton.js

It is not fully working as there is no backend involved yet but I d like still to merge as I am confident that the code will perform.
There some routes that need to be done with backend people, so we could have a chat about it.

Cheers,

### CODE REVIEW

File: src/Pages/FindPeople.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import React, { useState, useEffect } from 'react';
```
Comment (HelenHMaher) [inline src/Pages/FindPeople.js:1]: I see what you're doing here and I like the idea of returning the three most recently registered users if no match is found.  This is definitely something we could set up an api for.  I have a few questions though.

File: src/Pages/FindPeople.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        setSearchPeople(e.target.value)
+    }
+
+
+    return (
+        <React.Fragment>
+            <h2>Find friends!</h2>
+            <input type='text' name='searchFriends' onChange={handleChange} />
+            <button onClick={setSearchPeople}>Search</button>
+            {people && people.map(person => {
```
Comment (HelenHMaher) [inline src/Pages/FindPeople.js:67]: So here: your handleChange function is setting "searchPeople" everytime the user makes a new keystroke.  And every time "searchPeople" changes, useEffect will be called and send an axios request "/findProfile/${searchPeople}".  I think what you want is another local state [input, setInput] or something, where you can store whatever the user input is and then when they press the "Find Friends" button, onClick={e => setSearchPeople(e.target.value)} searchPeople is changed and the useEffect is triggered.  It's hard to test if it works without any data to send or search though -_-.

File: src/Pages/FriendsButton.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                } else if (data && data.acceptFriend === true) {
+                    setButtonText("Accept Friend Request")
+                } else if (data && data.cancelFrequest === true) {
+                    setButtonText("Cancel Friend Request")
+                } else {
+                    setError(true)
+                }
+            })
+    }, [])
+
```
Comment (HelenHMaher) [inline src/Pages/FriendsButton.js:43]: I was actually thinking about this functionality as well.  Something to discuss when we talk about the backend stuff.  Will be a good opportunity to chat with Olivia.  I think getting user data stored and being able to add/remove friends and search the user database is probably our first goal?

**Review by HelenHMaher — COMMENTED** [review]
I see what you're doing and it looks like a pretty solid place to start.

**Review by HelenHMaher — DISMISSED** [review]

File: src/Pages/FindPeople.js (hunk 1 above)
Comment (Edith2019) [reply src/Pages/FindPeople.js:1]: I would do it on load not if there is no reasult. Does that make sense?

File: src/Pages/FindPeople.js (hunk 2 above)
Comment (Edith2019) [reply src/Pages/FindPeople.js:67]: Okay, I can re-do the code but I think it should function like this.

File: src/Pages/FriendsButton.js (hunk 3 above)
Comment (Edith2019) [reply src/Pages/FriendsButton.js:43]: I don't really mind which one we decide to do first. I could actually write a draft on how to proceed in the db. I checked some MongoDB videos today to get familiar with it.

**Review by HelenHMaher — APPROVED** [review]

**Pull request merged**